lms/djangoapps/commerce/tests/test_signals.py

```diff
@@ -197,40 +207,71 @@ def test_notification_microsite(self, mock_is_request_in_microsite):  # pylint:
 
-        refund_ids = [1, 2, 3]
-        send_refund_notification(self.course_enrollment, refund_ids)
+        with mock.patch('commerce.signals.create_zendesk_ticket') as mock_zendesk:
+            refund_ids = [1, 2, 3]
+            send_refund_notification(self.course_enrollment, refund_ids)
+            body = generate_refund_notification_body(self.student, refund_ids)
+            mock_zendesk.assert_called_with(self.student.profile.name, self.student.email,
+                                            "[Refund] User-Requested Refund", body, ['auto_refund'])
 
-        # check headers and text content
-        self.assertEqual(
-            mock_email_class.call_args[0],
-            ("[Refund] User-Requested Refund", mock.ANY, self.student.email, ['payment@example.com']),
-        )
-        text_body = mock_email_class.call_args[0][1]
-        # check for a URL for each refund
-        for exp in [r'{0}/dashboard/refunds/{1}/'.format(TEST_PUBLIC_URL_ROOT, refund_id)
-                    for refund_id in refund_ids]:
-            self.assertRegexpMatches(text_body, exp)
-
-        # check HTML content
-        self.assertEqual(mock_message.attach_alternative.call_args[0], (mock.ANY, "text/html"))
-        html_body = mock_message.attach_alternative.call_args[0][0]
-        # check for a link to each refund
-        for exp in [r'a href="{0}/dashboard/refunds/{1}/"'.format(TEST_PUBLIC_URL_ROOT, refund_id)
-                    for refund_id in refund_ids]:
-            self.assertRegexpMatches(html_body, exp)
-
-        # make sure we actually SEND the message too.
-        self.assertTrue(mock_message.send.called)
+    def _mock_zendesk_api(self, status=201):
+        """ Mock Zendesk's ticket creation API. """
+        httpretty.register_uri(httpretty.POST, urljoin(ZENDESK_URL, '/api/v2/tickets.json'), status=status,
+                               body='{}', content_type=JSON)
+
+    def call_create_zendesk_ticket(self, name=u'Test user', email=u'user@example.com', subject=u'Test Ticket',
+                                   body=u'I want a refund!', tags=None):
+        """ Call the create_zendesk_ticket function. """
+        tags = tags or [u'auto_refund']
+        create_zendesk_ticket(name, email, subject, body, tags)
+
+    @override_settings(ZENDESK_URL=ZENDESK_URL, ZENDESK_USER=None, ZENDESK_API_KEY=None)
+    def test_create_zendesk_ticket_no_settings(self):
+        """ Verify the Zendesk API is not called if the settings are not all set. """
+        with mock.patch('requests.post') as mock_post:
+            self.call_create_zendesk_ticket()
+            self.assertFalse(mock_post.called)
+
+    def test_create_zendesk_ticket_request_error(self):
+        """ Verify exceptions are handled appropriately if the request to the Zendesk API fails. """
+        with mock.patch('requests.post', side_effect=Timeout) as mock_post:
+            self.call_create_zendesk_ticket()
+            self.assertTrue(mock_post.called)
```

Comment:
I assume what's being tested here is that the exception was caught...mind adding a comment or assertion to make that more explicit?